### DISCUSSION TITLE
ncTiledViewer - a loader and viewer of maps for the nCine

### DISCUSSION THREAD

**Discussion by encelo** [mapeditor/tiled#2952] [Show and tell]
I'm the developer of [nCine](https://ncine.github.io/), a cross-platform 2D game framework written in C++11 and optionally scriptable in Lua. It's in active development since June 2011 and it is released here on [GitHub](https://github.com/nCine/nCine) with a MIT license. :gear: 

My last project is a loader and viewer of Tiled maps that, I hope, will simplify game creation for users. :wink:
The project is called `ncTiledViewer` and it is able to load TMX files via [pujixml](https://pugixml.org/) and then display the map on the screen with nCine regular and mesh sprites. Currently it does only support the CSV tile layer format and thus it will not be able to load the official [examples](https://github.com/mapeditor/tiled/tree/master/examples) without a conversion. :disappointed:  

The project is not yet available on GitHub but I will update this discussion when it will be ready for prime time. :partying_face: 

**EDIT**
The project is now available on [GitHub](https://github.com/nCine/ncTiledViewer). You can [download](https://github.com/nCine/ncTiledViewer-artifacts/branches/all) a compiled version for one of the supported platforms or try it [online](https://ncine.github.io/nctiledviewer/).

![ncTiledViewer](https://user-images.githubusercontent.com/8441471/101704342-cf1fe480-3a84-11eb-8c51-bd74c220d634.png)

**Comment by bjorn** [comment]
Hey @encelo, welcome and thanks for showing us nCine and the ncTiledViewer, which looks already very functional!

If you can write your own engine I doubt adding support for decoding and decompressing tile layer data will be an issue, but right now CSV is the default layer data format used by Tiled so it may not be that much of a priority anyway. :slightly_smiling_face: 

I saw a great looking parallax scrolling demo video on your engine's website, so I thought you might also be interested to know that parallax scrolling factors are going to supported by Tiled very soon! (see #2951)

**Comment by encelo** [comment]
I have just published my [ncTiledViewer](https://ncine.github.io/2021-01-25-nctiledviewer/) project, you can find the source code on [GitHub](https://github.com/nCine/ncTiledViewer).

**Comment by encelo** [comment]
I have updated [ncTiledViewer](https://github.com/nCine/ncTiledViewer)!
You can now move around the map, rotate it or zoom it with the mouse, the keyboard or a gamepad.
You can compile or [download](<https://github.com/nCine/ncTiledViewer-artifacts/branches/all>) the project for Linux, macOS, Windows and Android or try it [online](https://ncine.github.io/nctiledviewer/).

**Comment by encelo** [comment]
I have updated the nCine and the `ncTiledViewer` project and there is now no latency in the ImGui overlay.
You can download the project or try it online at the usual addresses.
